demos/mqtt/mqtt_demo_basic_tls/mqtt_demo_basic_tls.c

```diff
@@ -497,39 +498,52 @@ static int handlePublishResend( MQTTContext_t * pMqttContext )
     int returnStatus = EXIT_SUCCESS;
     MQTTStatus_t mqttStatus = MQTTSuccess;
     uint8_t index = 0U;
+    MQTTStateCursor_t cursor = MQTT_STATE_CURSOR_INITIALIZER;
+    uint16_t packetIdToResend = 0;
 
+    assert( pMqttContext != NULL );
     assert( outgoingPublishPackets != NULL );
 
-    /* Resend all the QoS2 publishes still in the array. These are the
-     * publishes that hasn't received a PUBREC. When a PUBREC is
-     * received, the publish is removed from the array. */
-    for( index = 0U; index < MAX_OUTGOING_PUBLISHES; index++ )
+    /* For the sake of demonstrating how to use MQTT_PublishToResend(), the
+     * outgoingPublishPackets array is iterated through for the next pending
+     * publish packet to be resent. If the application requires increased
```

Comment:
The comment may not be clear to people with the context of the utility of the MQTT function and how it is used in this demo app. It would be more helpful to provide some information about what this function does. 
```suggestion
    /* The MQTT_PublishToResend() provides the next PUBLISH packet that should be resent keeping the message 
       * ordering preservation mentioned in the MQTT 3.1.1 specification. The provided publish packet is then searched in the
       * outgoingPublishPackets array to find it's associated PUBLISH message for attempting a re-publish operation.
```

Reply:
Sure